examples/envelope_designer/sketch.js

```diff
@@ -0,0 +1,103 @@
+var attackTime = 0.001
+var attackLevel = 1.0;
+var decayTime = 0.2;
+var decayLevel = 0.2;
+var releaseTime = 0.5;
+var releaseLevel = 0;
+var sustainTime = 0.5;
+
+var attackTimeInput, attackLevelInput;
+var decayTimeInput, decayLevelInput;
+var releaseTimeInput, releaseLevelInput;
+var sustainTimeInput;
+
+var maxTime = 1;
+var maxLevel = 5;
+var xScale = 100;
+
+var env, osc;
+
+function setup() {
+  var cnv = createCanvas(500, 500);
+  cnv.mousePressed(playEnv);
+
+  env = new p5.Env();  
+  osc = new p5.Oscillator('triangle');
+  osc.amp(env);
+  osc.start();
+  osc.freq(440);
+
+  attackTimeInput = createSlider(0, maxTime, attackTime, 0);
+  attackTimeInput.position(20, 15);
+  attackLevelInput = createSlider(0, maxLevel, attackLevel, 0);
+  attackLevelInput.position(160, 15);
+  decayTimeInput = createSlider(0, maxTime, decayTime, 0);
+  decayTimeInput.position(20, 45);
+  decayLevelInput = createSlider(0, maxLevel, decayLevel, 0);
+  decayLevelInput.position(160, 45);
+  sustainTimeInput = createSlider(0, maxTime, sustainTime, 0);
+  sustainTimeInput.position(20, 75);
+  releaseTimeInput = createSlider(0, maxTime, releaseTime, 0);
+  releaseTimeInput.position(20, 105);
+  releaseLevelInput = createSlider(0, maxLevel, releaseLevel, 0);
+  releaseLevelInput.position(160, 105);
```

Comment:
I think these UI elements need some labels, as it wasn't clear to me which slider controlled which parameter. We don't have a label element in p5.dom, but we should! https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label